enterprise_subsidy/apps/transaction/management/commands/write_reversals_from_enterprise_unenrollments.py

```diff
@@ -0,0 +1,216 @@
+                    f"{fulfillment_uuid}. Skipping Reversal creation for Transaction: {related_transaction}."
+                )
+                continue
+
+            # Continue on if no reversal found
+            logger.info(
+                f"{DRY_RUN_LOG_PREFIX}No existing Reversal found for enterprise fulfillment: {fulfillment_uuid}. "
+                f"Writing Reversal for Transaction: {related_transaction}."
+            )
+
+            # On initial release we are only supporting learner initiated unenrollments for OCM courses.
+            # OCM courses are identified by the lack of an external_reference on the Transaction object.
+            # Externally referenced transactions can be unenrolled through the Django admin actions related to the
+            # Transaction model.
+            if related_transaction.external_reference.exists():
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}Found unenrolled enterprise fulfillment: {fulfillment_uuid} related to "
+                    f"an externally referenced transaction: {related_transaction.external_reference.first()}. "
+                    "Skipping."
+                )
+                continue
+
+            # Memoize the content metadata for the course run fetched from the enterprise catalog
+            if not fetched_content_metadata.get(enrollment_course_run_key):
+                try:
+                    customer_uuid = related_transaction.ledger.subsidy.enterprise_customer_uuid
+                except Subsidy.DoesNotExist:
+                    logger.warning(
+                        f"{DRY_RUN_LOG_PREFIX}Unable to determine enterprise customer uuid for transaction: "
+                        f"{related_transaction}"
+                    )
+                    continue
+
+                content_metadata = ContentMetadataApi.get_content_metadata(
+                    customer_uuid,
+                    enrollment_course_run_key,
+                )
+                fetched_content_metadata[enrollment_course_run_key] = content_metadata
+            else:
+                content_metadata = fetched_content_metadata.get(enrollment_course_run_key)
+
+            # Check if the OCM unenrollment is refundable
+            if not self.unenrollment_can_be_refunded(content_metadata, enterprise_course_enrollment):
+                logger.info(
+                    f"{DRY_RUN_LOG_PREFIX}Unenrollment from course: {enrollment_course_run_key} by user: "
+                    f"{enterprise_course_enrollment.get('enterprise_customer_user')} is not refundable."
+                )
+                continue
+
+            logger.info(
+                f"{DRY_RUN_LOG_PREFIX}Course run: {enrollment_course_run_key} is refundable for enterprise "
+                f"customer user: {enterprise_course_enrollment.get('enterprise_customer_user')}. Writing "
+                "Reversal record."
+            )
+
+            if not dry_run:
```

Comment:
you may be able to move the idempotency key generation into this block, to keep it closer to where you're actually going to use it. this co-location makes it easier to read/understand.